[flutter_tools] support files in flutter analyze #96231

Jasguerrero opened the pull request:
https://github.com/flutter/flutter/issues/96231
"analyze" command only accepted directories now you can pass files to it

Jasguerrero committed:
[flutter-tools] pass files to analyze command (#96231)

Jasguerrero committed:
Test with more than one file with errors (#96231)

christopherfujino commented:
@Jasguerrero you will have to rebase upstream in order to get the updated names of the `tool_integration_tests`. This is a breaking change I made to the infrastructure unrelated to your test. Rebasing will be something like `git fetch upstream && git rebase upstream/master`.

Jasguerrero committed:
Merge branch 'master' into pass_files_to_analyze

christopherfujino commented on packages/flutter_tools/test/integration.shard/analyze_once_test.dart:
```diff
@@ -100,12 +100,27 @@ void main() {
+  testWithoutContext('passing more than one file with errors', () async {
+    final File secondFile = fileSystem.file(fileSystem.path.join(projectPath, 'lib', 'second.dart'))
+      ..createSync(recursive: true)
+      ..writeAsStringSync(r'''import 'package:flutter/material.dart''');
```
```suggestion
      ..writeAsStringSync(r"import 'package:flutter/material.dart");
```

Using double quotes makes the syntax error more obvious.

christopherfujino commented on packages/flutter_tools/test/integration.shard/analyze_once_test.dart:
```diff
@@ -100,12 +100,27 @@ void main() {
 
-  // testWithoutContext a specific file outside the current directory
-  testWithoutContext('passing one file throws', () async {
+  testWithoutContext('passing one file works', () async {
```
These tests look good, can you add two more test cases: `'passing one file with errors are detected'` and `'passing more than one file without errors works'`?

Jasguerrero replied:
Sure

christopherfujino commented on packages/flutter_tools/lib/src/commands/analyze_once.dart:
```diff
@@ -48,17 +48,15 @@ class AnalyzeOnce extends AnalyzeBase {
 
-    // find directories from argResults.rest
-    final Set<String> directories = Set<String>.of(argResults.rest
+    // find directories or files from argResults.rest
```
does the dartanalyzer support mixing and matching files with directories?

Jasguerrero replied:
Yeah seems like it support mixing I will also add a test for that case

Jasguerrero committed:
new tests for analyze_once

Add new tests with more than one file with errors and mixing directories and files

christopherfujino reviewed (approved):
LGTM!

zanderso reviewed (approved):
lgtm w/ nit

Jasguerrero committed:
replace type of item to name and add test #96231
